bracket-push.json

```diff
@@ -0,0 +1,49 @@
+{
+  "cases": [
+    {
+      "description": "empty string",
+      "input": "",
+      "expected": true
+    },
+    {
+      "description": "unpaired brackets",
+      "input": "[[",
+      "expected": false
+    },
+    {
+      "description": "wrong ordered brackets",
+      "input": "}{",
+      "expected": false
+    },
+    {
+      "description": "paired and nested brackets",
+      "input": "([{}({}[])])",
+      "expected": true
+    },
+    {
+      "description": "unpaired and nested brackets",
+      "input": "([{])",
+      "expected": false
+    },
+    {
+      "description": "paired and wrong nested brackets",
+      "input": "[({]})",
+      "expected": false
+    },
+    {
+      "description": "paired with whitespace",
+      "input": "{ }",
+      "expected": true
+    },
+    {
+      "description": "math expression",
+      "input": "(((185 + 223.85) * 15) - 543)/2",
+      "expected": true
+    },
+    {
+      "description": "complex latex expression",
+      "input": "\right(\begin{array}{cc} \frac{1}{3} & x^2 \\end{array})",
```

Comment:
No warnings because there is nothing that you are doing 'wrong'.  It is just taking what you give it, and working with it, without any way to know you wanted something different.